modules/ding_gatewayf/includes/ding_gatewayf.admin.inc

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @file
+ * Defines the administration interface for the module.
+ */
+
+/**
+ * Implements hook_form().
+ */
+function ding_gatewayf_admin_settings_form() {
+  $form = array();
+  $defaults = variable_get('ding_gatewayf', array());
```

Comment:
Should this use `_ding_gatewayf_load_configuration()` instead? This would also reduce `#default_value` complexity.